### PR TITLE
Don't depend on weak references for finite tasks

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/FiniteTask.scala [hunk 1]
```diff
@@ -18,14 +16,15 @@ trait FiniteTaskFinishedListener {
...
+  private var listener: Option[FiniteTaskFinishedListener] = None
 
   /**
     * Adds a task finish listener to listen to this task.
     * @param listener the listener of finish events
     */
-  def addFinishedListener(listener: FiniteTaskFinishedListener): Unit = {
-    listeners = WeakReference(listener) :: listeners
+  def setFinishedListener(newListener: FiniteTaskFinishedListener): Unit = {
+    assert(listener == None)
```
Comment (kunalsheth) [inline core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/FiniteTask.scala:26]: Can we also print something incase the `-ea` flag is not set?

**Review by kunalsheth — DISMISSED** [review]

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/FiniteTask.scala (hunk 1 above)
Comment (shadaj) [reply core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/FiniteTask.scala:26]: The way Scala's assert is implemented, exceptions will be thrown no matter what the flag is set to.

**Comment by shadaj** [comment]
@kunalsheth ping re-review?

**Review by PhilipAxelrod — APPROVED** [review]
Lgtm

**Pull request merged**